lib/apollo_tracing/middleware.ex

```diff
@@ -3,52 +3,38 @@ defmodule ApolloTracing.Middleware do
   Documentation for ApolloTracing.
   """
 
-  alias ApolloTracing.{Schema, Schema.Execution, Schema.Execution.Resolver}
+  alias ApolloTracing.Schema.Execution.Resolver
   alias Absinthe.Resolution
 
   # Called before resolving
-  def call(%Resolution{state: :unresolved} = res, _config) do
-    %{acc: %{
-      apollo_tracing_start_time: start_mono_time,
-      apollo_tracing: %Schema{
-        execution: %Execution{
-          resolvers: resolvers_so_far
-        }
-      }
-    }} = res
-
+  # if there isn't an `ApolloTracing` flag set then we aren't actually doing any tracing
+  def call(%Resolution{acc: %{apollo_tracing_start_time: start_mono_time}, state: :unresolved} = res, _config) do
     now = System.monotonic_time()
     resolver = %Resolver{
       path: Absinthe.Resolution.path(res),
       parentType: res.parent_type.name,
       fieldName: res.definition.name,
       returnType: Absinthe.Type.name(res.definition.schema_node.type, res.schema),
       startOffset: now - start_mono_time,
-      duration: nil
     }
 
-    res = put_in(
-      res.acc.apollo_tracing.execution.resolvers,
-      [resolver | resolvers_so_far]
-    )
-
-    %{res | middleware:
-       res.middleware ++ [{{__MODULE__, :after_field}, [start_time: now]}]
+    %{res |
+      extensions: Map.put(res.extensions, __MODULE__, resolver),
```

Comment:
In the previous way this worked every `call` would basically handle the last item on the `execution.resolvers` stack. Problems occur with the following type of situation:

```
{
  foo1: getFoo(id: 1) { name }
  foo2: getFoo(id: 2) { name }
}
```

If the `getFoo` field is using batching, then `call/2` in this module will run for `foo1` field, but then `foo1` gets suspended. Absinthe moves on to `foo2` BEFORE `foo1` completes, and thus before the `after_field` call happens on `foo1`. This causes a variety of problems.

In any case, I solved it by storing the `resolver` info under the extensions key of the resolution struct. This extensions key is not automatically aggregated, and thus is a handy spot to store data you want to just keep for this specific field. I store it under the `__MODULE__` key just to avoid name conflicts.